trie/database.go

```diff
@@ -697,7 +697,7 @@ func (db *Database) Cap(limit common.StorageSize) error {
 //
 // Note, this method is a non-synchronized mutator. It is unsafe to call this
 // concurrently with other mutators.
-func (db *Database) Commit(node common.Hash, report bool) error {
+func (db *Database) Commit(node common.Hash, report bool, callback func(common.Hash)) error {
```

Comment:
What's the purpose of this callback?

Reply:
Because we want to prune CHT and Bloom Trie. Since the trie nodes of the latest section are enough for generating next Merkle root. So I use the callback to collect all trie nodes of the current section and delete all other.